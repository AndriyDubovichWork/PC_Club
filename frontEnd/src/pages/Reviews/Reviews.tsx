import React from 'react';
import SubHeader from '../../Shared/SubHeader/SubHeader';
import style from './Reviews.module.scss';
import Banner from '../../Shared/Banner/Banner';
import Testimonials from '../../Shared/Testimonials/Testimonials';

const Reviews = () => {
  return (
    <>
      <main className={style.Reviews}>
        <SubHeader title='Reviews' />
        <Testimonials />
      </main>
      <Banner />
    </>
  );
};

export default Reviews;
