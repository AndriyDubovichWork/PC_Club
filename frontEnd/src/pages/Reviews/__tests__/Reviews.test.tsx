import { render, screen, cleanup } from '@testing-library/react';
import Reviews from '../Reviews';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('Reviews match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/reviews']}>
        <Routes>
          <Route path='/PC_Club/reviews' element={<Reviews />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Reviews contains main tag', () => {
  const { getByRole } = render(
    <MemoryRouter initialEntries={['/PC_Club/reviews']}>
      <Routes>
        <Route path='/PC_Club/reviews' element={<Reviews />} />
      </Routes>
    </MemoryRouter>
  );

  expect(getByRole('main')).toBeInTheDocument();
});
