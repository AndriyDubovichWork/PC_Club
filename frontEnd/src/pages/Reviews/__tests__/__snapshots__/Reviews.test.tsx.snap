// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews match snapshot 1`] = `
Array [
  <main
    className="Reviews"
  >
    <img
      className="backGroundPC"
      loading="lazy"
    />
    <div
      className="Text"
    >
      <h1
        className="Title"
      >
        Reviews
      </h1>
      <nav
        aria-label="breadcrumb"
        className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            className="MuiBreadcrumbs-li"
          >
            <a
              aria-current="page"
              className="Link active"
              href="/PC_Club"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            aria-hidden={true}
            className="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            className="MuiBreadcrumbs-li"
          >
            <a
              aria-current="page"
              className="Link active"
              href="/PC_Club/reviews"
              onClick={[Function]}
            >
              Reviews
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      className="Reviews"
    >
      <div
        className="Content"
      >
        <h3
          className="preTitle"
        >
          Reviewed by People
        </h3>
        <h1
          className="Title"
        >
          Client's Testimonials
        </h1>
        <h4
          className="subTitle"
        >
          Discover the positive impact we've made on the our clients by reading through their testimonials. Our clients have experienced our service and results, and they're eager to share their positive experiences with you.
        </h4>
        <div
          className="Testimonials"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-er2eq5-MuiPaper-root"
          >
            <div>
              <span
                className="Text"
              >
                "
                I had a great experience at PC Club! The staff was friendly and helpful. I highly recommend this store.
                "
              </span>
            </div>
            <div
              className="Author"
            >
              <img
                alt="testimonial"
                className="Img"
                loading="lazy"
                src="harry.jpg"
              />
              <h3>
                Harry Potter
              </h3>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-19pa9qe-MuiSvgIcon-root"
                data-testid="FormatQuoteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-er2eq5-MuiPaper-root"
          >
            <div>
              <span
                className="Text"
              >
                "
                PC Club's great selection and knowledgeable staff make them my go-to for tech needs.
                "
              </span>
            </div>
            <div
              className="Author"
            >
              <img
                alt="testimonial"
                className="Img"
                loading="lazy"
                src="Ron.jpg"
              />
              <h3>
                Ron Weasley
              </h3>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-19pa9qe-MuiSvgIcon-root"
                data-testid="FormatQuoteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>,
  <div
    className="Banner"
  >
    <h1
      className="Title"
    >
      Save big with our cheap PC Club
    </h1>
    <h3
      className="subTitle"
    >
      Top PCs. Global Suppliers.
      <span
        className="HighLighted"
      >
         24/7
      </span>
       Support.
    </h3>
  </div>,
]
`;
